File: Interviewbit/Bit Manipulation/xor-ing-the-subarrays.cpp
int Solution::solve(vector<int> &A) {
    int n=A.size();
    int ans=0;
    for(int i=0;i<A.size();i++){
        int leftele=i;
        int rightele=n-i;
        int freq=i*(n-i)+(n-i);
        if(freq%2==0)  continue;
        else{
            ans=ans^A[i];
        }
    }
    return ans;
}

File: Interviewbit/Bit Manipulation/xor-ing-the-subarrays.java
public class Solution {
    public int solve(ArrayList<Integer> A) {
        int ans=0;
        int n=A.size();
        for(int i=0;i<n;i++){
            int left=i;
            int right=n-i;
            int freq=i*(n-i)+(n-i);
            if(freq%2!=0){
                ans=ans^A.get(i);
            }
        }
        return ans;
    }
}
